src/transformers/models/t5/modeling_t5.py

```diff
@@ -513,7 +513,7 @@ def project(hidden_states, proj_layer, key_value_states, past_key_value):
         if position_bias is None:
             if not self.has_relative_attention_bias:
                 position_bias = torch.zeros(
-                    (1, self.n_heads, real_seq_length, key_length), device=scores.device, dtype=scores.dtype
+                    (1, self.n_heads+len(self.pruned_heads), real_seq_length, key_length), device=scores.device, dtype=scores.dtype
```

Comment:
Hi, in this line 

https://github.com/CaffreyR/transformers/blob/main/src/transformers/models/t5/modeling_t5.py#L369 

we have update the `self.n_heads`, but in the `forward`, we have to keep `position_bias` to remain `12`, so that we use mask to do the prune progress.

https://github.com/CaffreyR/transformers/blob/main/src/transformers/models/t5/modeling_t5.py#L531

Reply:
You can see that after `prune_head` in T5 attention, the `self.n_heads` has already been the remaining heads number. But in the `forward` we have to tell the T5 attention the shape of `position_bias`